presto/dataset.py

```diff
@@ -192,20 +258,31 @@ def split_df(
             is_train = ~df.end_date_ts.dt.year.isin(val_years)
 
         logger.info(f"Using {len(is_val) - sum(is_val)} train and {sum(is_val)} val samples")
-        train = df[is_train]
-        val = df[is_val]
-        return train, val
+
+        return df[is_train], df[is_val]
 
 
 class WorldCerealMaskedDataset(WorldCerealBase):
-    def __init__(self, dataframe: pd.DataFrame, mask_params: MaskParamsNoDw):
+    def __init__(
+        self,
+        dataframe: pd.DataFrame,
+        mask_params: MaskParamsNoDw,
+        task_type: str = "cropland",
+        croptype_list: List = [],
+        model_mode: str = "",
```

Comment:
Am I right in understanding that `model_mode` only takes two values - `"Hierarchical CatBoostClassifier"` or `""`?

If so can this be replaced with a boolean? e.g. `return_hierarchical_labels: bool = False`? This would make it clearer what this parameter controls

Reply:
hmmm, good idea 🤔 
my thinking was that `model_mode` parameter already existed, and thus can be just passed down from the the very top... 

I use it here when I iterate over the possible model_modes (`Regression, Random Forest, CatBoostClassifier, Hierarchical CatBoostClassifier`), and pass this parameter to `DataLoader` so that I can get correct targets from it. 
https://github.com/WorldCereal/presto-worldcereal/blob/d72d8199d07a50bcfc2f01075ef75f5229f16bc6/presto/eval.py#L231

If you can suggest more elegant way to do it without dragging this `model_mode` that far, I'd be happy to try it. I don't it see how to overcome it now 🙈

Reply:
Potential solution - in the dataset init, have the argument:

```python
    return_hierarchical_labels: bool = False
```
and then when you call the dataset:

```python
   DataLoader(..., return_hierarchical_labels=True if "Hierarchical" in model_mode else False)
```